src/functions/src/main/java/org/apache/jmeter/functions/TimeFunction.java

```diff
@@ -96,13 +99,17 @@ public String execute(SampleResult previousResult, Sampler currentSampler) throw
                 long div = Long.parseLong(fmt.substring(1)); // should never case NFE
                 datetime = Long.toString(System.currentTimeMillis() / div);
             } else {
-                if (fmt.contains("u")) {
+                DateTimeFormatter df = DateTimeFormatter // Not synchronised, so can't be shared
+                        .ofPattern(fmt);
```

Comment:
Is it really non-thread-safe?

The javadoc says otherwise: https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

> A formatter created from a pattern can be used as many times as necessary, it is immutable and is thread-safe.